storage/deposit/store.go

```diff
@@ -22,144 +22,107 @@ package deposit
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"sync"
 
-	sdkcollections "cosmossdk.io/collections"
-	"cosmossdk.io/core/store"
 	ctypes "github.com/berachain/beacon-kit/consensus-types/types"
-	"github.com/berachain/beacon-kit/errors"
 	"github.com/berachain/beacon-kit/log"
-	"github.com/berachain/beacon-kit/storage"
-	"github.com/berachain/beacon-kit/storage/encoding"
+	"github.com/berachain/beacon-kit/primitives/common"
+	depositstorev1 "github.com/berachain/beacon-kit/storage/deposit/v1"
+	dbm "github.com/cosmos/cosmos-db"
 )
 
-const KeyDepositPrefix = "deposit"
+const (
+	unset uint8 = 0
+	V1    uint8 = 1
+)
 
-// KVStore is a simple KV store based implementation that assumes
-// the deposit indexes are tracked outside of the kv store.
-type KVStore struct {
-	store sdkcollections.Map[uint64, *ctypes.Deposit]
+type Store interface {
+	GetDepositsByIndex(ctx context.Context, startIndex uint64, depRange uint64) (ctypes.Deposits, common.Root, error)
+	EnqueueDeposits(ctx context.Context, deposits []*ctypes.Deposit) error
+	Prune(ctx context.Context, start, end uint64) error
+	Close() error
+}
 
-	// closeFunc is a closure that closes the underlying database
-	// used by store to ensure that all writes are flushed to disk.
-	// We guarantee that closeFunc is called at maximum only once.
-	closeFunc CloseFunc
-	once      sync.Once
+type StoreManager interface {
+	Store
+}
```

Comment:
for the time being `StoreManager` interface seems redundant. However as soon as we will introduce deposit store v2, it will be useful to handle data migration logic. So proactivelly using it in the rest of the codebase in stead of Store